dataflow/gemma-flex-template/README.md

```diff
@@ -0,0 +1,205 @@
+
+### Create a custom container
+
+To build a custom container, use Docker. This repository contains a Dockerfile that you can use to build your custom container. To build and push a container to Artifact Registry by using DockerFollow, follow the instructions in the [Build and push the image](https://cloud.google.com/dataflow/docs/guides/build-container-image#build_and_push_the_image) section of "Build custom container images for Dataflow" in the Google Cloud documentation.
+
+### Create Pub/Sub topics for input and output
+
+To create your Pub/Sub source and sink, follow the instructions in [Create a Pub/Sub topic](https://cloud.google.com/pubsub/docs/create-topic#pubsub_create_topic-Console) in the Google Cloud documentation. For this example, create two topics, one input topic and one output topic. For input, the topic must have a subscription that you can provide to the Gemma model. 
+
+### Download and save the model
+
+Save a version of the Gemma 2B model. Downloaded the model from [Kaggle](https://www.kaggle.com/models/google/gemma/pyTorch/1.1-2b-it). This download is a `.tar.gz` archive. Extract the archive into a directory and name it `pytorch_model`. 
+
+### Optional: Create a new virtual environment
+
+The Python major and minor version contained in the custom container must match the environment used for job submission. For this example, use Python version 3.10.
+
+```
+python3.10 -m venv /tmp/venv
+source /tmp/venv/bin/activate
+```
+
+For more information, see [venv — Creation of virtual environments](https://docs.python.org/3/library/venv.html).
+
+### Install dependencies
+
+Install Apache Beam and the dependencies required to run the pipeline in your local environment. 
+
+```
+pip install -U -r requirements.txt
+```
+
+This example also requires access to the base class for the PyTorch implementation of Gemma.
+```
+git clone https://github.com/google/gemma_pytorch.git
+pip install ./gemma_pytorch
+```
+
+## Code overview
+
+This section provides details about the custom model handler and the formatting `DoFn` used in this example.
+
+### Custom model handler
+This notebook defines a custom model handler that loads the model. The model handler constructs a configuration object and loads the model's checkpoint from the local filesystem.
+Because this approach differs from the PyTorch model loading process followed in the Beam PyTorch model handler, a custom implementation is necessary. 
+
+To customize the behavior of the handler, implement the following methods: `load_model`, `validate_inference_args`, and `share_model_across_processes`.
+
+The PyTorch implementation of the Gemma models has a `generate` method
+that generates text based on a prompt. To route the prompts correctly, use this function in the `run_inference` function.
+
+```py
+class GemmaPytorchModelHandler(ModelHandler[str, PredictionResult,
```

Comment:
having large number of code in the md. might cause an inconsistency down the road if we modify the source but not the MD. Maybe we only highlight most important points here, and send user to the source instead? We can and should have comments in the source anyway.

Reply:
This is a fair point, I think the emphasis in the text is on model loading being different so I think cutting the large model function and the `run_inference` call itself is fine for simplicity's sake